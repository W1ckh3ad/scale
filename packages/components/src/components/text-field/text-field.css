scale-text-field {
  --transition: all var(--scl-motion-duration-fast)
    var(--scl-motion-easing-standard);
  --radius: var(--scl-radius-4);
  --border: var(--scl-spacing-1) solid var(--scl-color-text-standard);
  --border-error: var(--scl-spacing-2) solid var(--scl-color-background-error);
  --border-color-hover: var(--scl-color-primary-hover, #f90984);
  --border-color-focus: var(--scl-color-primary-hover, #f90984);
  --box-shadow-focus: 0 0 0 var(--scl-spacing-2) var(--scl-color-focus);
  --height: var(--scl-spacing-48);
  --height-small: var(--scl-spacing-40);
  --spacing-x: var(--scl-spacing-12);
  --color-disabled: var(--scl-color-background-disabled);
  --background-disabled: var(--scl-color-white);

  /*meta*/
  --font-weight-meta: var(--scl-font-weight-bold);
  --font-size-meta: var(--scl-font-size-12);
  --line-height-meta: var(--scl-font-line-height-133);
  --spacing-y-meta: var(--scl-spacing-4);
  --color-meta-error: var(--scl-color-text-error);

  /*input*/
  --spacing-input: var(--scl-spacing-12) var(--spacing-x) 0
    calc(var(--spacing-x) - 1px);
  --transition-input: var(--transition);
  --font-size-input: var(--scl-font-size-16);

  /*counter*/
  --transition-counter: var(--transition);
  --font-size-counter: var(--font-size-meta);
  --line-height-counter: var(--line-height-meta);
  --color-counter-error: var(--color-meta-error);

  /*helper-text*/
  --transition-helper-text: var(--transition);
  --font-size-helper-text: var(--font-size-meta);
  --line-height-helper-text: var(--line-height-meta);
  --color-helper-text: var(--scl-color-blue-70);
  --color-helper-text-error: var(--color-meta-error);

  /*placeholder*/
  --transition-placeholder: var(--transition);
  --color-placeholder: var(--scl-color-grey-60);

  /*label*/
  --color-label: var(--scl-color-grey-60);
  --z-index-label: var(--scl-z-index-10);
  --transition-label: var(--transition);
  --font-size-label: var(--scl-font-size-16);
  --font-weight-label: var(--scl-font-weight-medium);
  --font-size-label-small: var(--scl-font-size-16);
  --font-weight-label-small: var(--scl-font-weight-medium);
  --font-size-label-small-focus: var(--scl-font-size-10);
  --font-size-label-focus: var(--scl-font-size-10);
  --font-weight-label-focus: var(--scl-font-weight-bold);
}

.text-field {
  position: relative;
}
.text-field .input__helper-text,
.text-field .input__counter {
  font-weight: var(--font-weight-meta);
}
.text-field .input__input {
  width: 100%;
  height: var(--height);
  margin: 0;
  display: flex;
  outline: none;
  padding: var(--spacing-input);
  z-index: 1;
  box-sizing: border-box;
  transition: var(--transition-input);
  font-family: inherit;
  font-size: var(--font-size-input);
  border-radius: var(--radius);
  border: var(--border);
}
.text-field .input__counter {
  display: flex;
  transition: var(--transition-counter);
  margin-left: auto;
  padding-right: var(--spacing-x);
  justify-content: flex-end;
  font-size: var(--font-size-counter);
  line-height: var(--line-height-counter);
  color: inherit;
}
.text-field .input__helper-text {
  transition: var(--transition-helper-text);
  padding-left: var(--spacing-x);
  font-size: var(--font-size-helper-text);
  line-height: var(--line-height-helper-text);
  color: var(--color-helper-text);
}
.text-field .input__meta {
  display: flex;
  justify-content: space-between;
  margin-top: var(--spacing-y-meta);
}

.text-field:not(.text-field--disabled) .input__input:hover {
  border-color: var(--border-color-hover);
}

.text-field:not(.text-field--disabled) .input__input:focus {
  border-color: var(--border-color-focus);
  box-shadow: var(--box-shadow-focus);
}

.text-field:not(.text-field--disabled) .input__input:focus::placeholder {
  color: var(--color-placeholder);
}

.text-field .input__input::placeholder,
.text-field ::placeholder {
  color: transparent;
  transition: var(--transition-placeholder);
}

.input__label {
  top: 0;
  left: 0;
  color: var(--color-label);
  display: flex;
  z-index: var(--z-index-label);
  position: absolute;
  transition: var(--transition-label);
  pointer-events: none;
  font-size: var(--font-size-label);
  transform: translate(
    var(--spacing-x),
    calc((var(--height) - var(--font-size-label)) / 2)
  );
  font-weight: var(--font-weight-label);
}
.text-field--has-focus .input__label,
.animated .input__label {
  transform: translate(var(--spacing-x), var(--scl-spacing-8));
  font-size: var(--font-size-label-focus);
  font-weight: var(--font-weight-label-focus);
}
.text-field--status-error .input__input {
  border: var(--border-error);
}
.text-field--status-error .input__helper-text {
  color: var(--color-helper-text-error);
}
.text-field--status-error .input__counter {
  color: var(--color-counter-error);
}
.text-field--size-small .input__input {
  height: var(--height-small);
}
.text-field--size-small .input__label {
  font-size: var(--font-size-label-small);
  transform: translate(
    var(--spacing-x),
    calc((var(--height-small) - var(--font-size-label-small)) / 2)
  );
  font-weight: var(--font-weight-label-small);
}
.text-field--size-small.text-field--has-focus .input__label,
.text-field--size-small.animated .input__label {
  transform: translate(var(--spacing-x), var(--scl-spacing-4));
  font-size: var(--font-size-label-small-focus);
}
.text-field--transparent .input__input {
  background-color: transparent;
}

.text-field--disabled label,
.text-field--disabled .input__label,
.text-field--disabled input,
.text-field--disabled .input__input,
.text-field--disabled .input__meta,
.input.text-field--disabled .input__counter,
.text-field--disabled .input__helper-text {
  cursor: not-allowed;
  border-color: var(--color-disabled);
  color: var(--color-disabled);
  background: var(--scl-color-white);
}
.text-field--disabled.animated label.input__label {
  color: var(--color-disabled);
}

/*Hide Arrows From Input Number*/

/* Chrome, Safari, Edge, Opera */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

/* Firefox */
input[type='number'] {
  -moz-appearance: textfield;
}
