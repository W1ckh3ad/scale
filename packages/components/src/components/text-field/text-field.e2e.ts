import { newE2EPage } from '@stencil/core/testing';

describe('scale-text-field', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-text-field/>');
    const element = await page.find('scale-text-field');
    expect(element).toHaveClass('hydrated');
  });
});
