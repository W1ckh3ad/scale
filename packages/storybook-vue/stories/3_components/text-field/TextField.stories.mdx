import { Meta, Story, ArgsTable, Canvas, Description } from "@storybook/addon-docs/blocks";
import ScaleTextField from "./ScaleTextField.vue";
import { action } from "@storybook/addon-actions";
export const ICON = "M16.471,8.321 L12.613594,0.502886112 L8.755,8.321 L0.126435258,9.57533801 L6.37,15.66 L4.89610548,24.2548735 L12.613,20.197 L20.3310826,24.2548735 L18.856,15.66 L25.1007528,9.57533801 L16.471,8.321 Z M9.41891527,9.23556376 L12.613,2.761 L15.8082728,9.23556376 L22.951,10.273 L17.7826928,15.3122039 L19.003,22.427 L12.613594,19.067774 L6.223,22.427 L7.44449521,15.3122039 L2.275,10.273 L9.41891527,9.23556376 Z";

<Meta
  title="Components/Text Field"
  component={ScaleTextField}
  argTypes={{
    resize: {
      control: {
        type: "select",
        options: ["unset", "none", "vertical", "horizontal"]
      }
    },
    type: {
      control: {
        type: "select",
        options: [
          "email",
          "hidden",
          "number",
          "password",
          "tel",
          "text",
        ]
      }
    }
  }}
/>

export const Template = (args, { argTypes }) => ({
  components: { ScaleTextField },
  props: {
    label: String,
    ...ScaleTextField.props
  },
  template: `
    <scale-text-field
      :name="name"
      :type="type"
      :label="label"
      :helper-text="helperText"
      :status="status"
      :max-length="maxLength"
      :min-length="minLength"
      :placeholder="placeholder"
      :disabled="disabled"
      :required="required"
      :counter="counter"
      :size="size"
      :value="value"
      :input-id="inputId"
      @scaleChange="scaleChange"
      @scaleFocus="scaleFocus"
      @scaleBlur="scaleBlur"
      @scaleKeyDown="scaleKeyDown"
    ></scale-text-field>
  `,
  methods: {
    scaleChange: action("scaleChange"),
    scaleFocus: action("scaleFocus"),
    scaleBlur: action("scaleBlur"),
    scaleKeyDown: action("scaleKeyDown"),
  }
})

<div style={{display: 'inline-flex', alignItems: 'center', justifyContent: 'space-between', width: '100%'}}>
    <h1>Text Field</h1>
    <img src="assets/aa.png" alt="Accessible AA" />
</div>

## Standard

<Canvas withSource="none">
  <Story
    name="Standard"
    args={{
      label: "An input"
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Standard" />


## Text With Events

<Canvas withSource="none">
  <Story
    name="Text With Events"
    args={{
      label: "Text With Events"
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-text-field @scaleChange="handleChange" @scaleBlur="handleBlur" @scaleFocus="handleFocus" @scaleKeyDown="handleKeyDown"></scale-text-field>
```


## Placeholder

<Canvas withSource="none">
  <Story
    name="Placeholder"
    args={{
      label: "Placeholder",
      placeholder: "this is the placeholder"
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-text-field label="Placeholder" placeholder="this is the placeholder"></scale-text-field>
```


## Helper Text

<Canvas withSource="none">
  <Story
    name="Helper Text"
    args={{
      label: "Helper Text",
      helperText: "Make sure to fill this"
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-text-field label="Helper Text" helperText="Make sure to fill this"></scale-text-field>
```


## Small

<Canvas withSource="none">
  <Story
    name="Small"
    args={{
      label: "Small",
      size: 'small'
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-text-field label="Small" size="small"></scale-text-field>
```


## With Error

<Canvas withSource="none">
  <Story
    name="With Error"
    args={{
      label: "With Error",
      status: "error",
      helperText: "something is wrong"
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-text-field label="With Error" helper-text="something is wrong" status="error"></scale-text-field>
```


## Disabled

<Canvas withSource="none">
  <Story
    name="Disabled"
    args={{
      label: "Disabled",
      disabled: true
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-text-field label="Disabled" disabled></scale-text-field>
```


## Max Length With Counter

<Canvas withSource="none">
  <Story
    name="Max Length With Counter"
    args={{
      label: "Max Length With Counter",
      maxLength: 10,
      counter: true
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-text-field label="Max Length With Counter" max-length="10" counter></scale-text-field>
```
