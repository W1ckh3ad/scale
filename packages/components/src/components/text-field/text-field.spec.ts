import { newSpecPage } from '@stencil/core/testing';
import { TextField } from './text-field';
describe('TextField', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [TextField],
      html: `<scale-text-field></scale-text-field>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
