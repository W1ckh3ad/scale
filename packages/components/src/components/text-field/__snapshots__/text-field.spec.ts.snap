// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should match snapshot 1`] = `
<scale-text-field>
  <div class="input input--type-text">
    <label class="input__label" htmlfor="input-text-field0"></label>
    <input aria-describedby="helper-message-1" class="input__input" id="input-text-field0" type="text" value="">
  </div>
</scale-text-field>
`;
